### PR TITLE
AWS integration

### PR DESCRIPTION
**Pull request by x-wf**

### CODE REVIEW

File: tools/libcloud/aws.py [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+    params["LaunchTemplate.LaunchTemplateName"] = config.STORAGE["AWS_CORE_TEMPLATE_CONFIG"]["LaunchTemplateName"]
+    params["MinSize"] = amount
+    params["MaxSize"] = amount
+    try:
+        # use auto scaling driver
+        response = as_driver.connection.request("/", params=params)
+
+        logging.info("- Core group for %s has been created.", region)
+    except BaseHTTPError as e:
+        logging.info("An error occured communicating with autoscaling service.")
```
Comment (swiec) [inline tools/libcloud/aws.py]: better to pass original error message

File: tools/libcloud/aws.py [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+
+    params["AutoScalingGroupName"] = "mtps-cores-{0}".format(driver.region_name)
+    params["ForceDelete"] = "true"
+
+    try:
+        # use auto scaling driver
+        response = as_driver.connection.request("/", params=params)
+        logging.info("- Core group for %s has been destroyed.", region)
+    except BaseHTTPError as e:
+        logging.error("An error occured communicating with autoscaling service.")
```
Comment (swiec) [inline tools/libcloud/aws.py]: as above

File: explorer/nginx/conf.d/nginx.conf.envsubst [hunk 3]
```diff
@@ -70,6 +70,19 @@ http {
             add_header Cache-Control "public, max-age=1";
             include conf.d/enable-cors.conf;
         }
+
+         # expose current universe for resources to fetch
```
Comment (swiec) [inline explorer/nginx/conf.d/nginx.conf.envsubst:74]: What this change is for? I don't really get it

**Review by swiec — COMMENTED** [review]
Good to go for me.

File: tools/libcloud/aws.py (hunk 1 above)
Comment (x-wf) [reply tools/libcloud/aws.py]: Completely agreed, unfortunately `e` comes empty all the time, we're not handling it properly with libcloud error handling.

We could create a task for this 👍

File: explorer/nginx/conf.d/nginx.conf.envsubst (hunk 3 above)
Comment (x-wf) [reply explorer/nginx/conf.d/nginx.conf.envsubst:74]: This is regarding the architecture.
Unfortunatly, by filling in the `CORE_UNIVERSE` inside the `core` cloud-init file, it was impossible to transfer it over the AWS API.

So, architecture was changed to make the explorer provide the universe, and the core nodes grab the universe from an endpoint.

**Pull request merged**